crates/loader/src/validation.rs

```diff
@@ -1,19 +1,37 @@
-use anyhow::{anyhow, Result};
-
-const ONLY_ALLOWED_KV_STORE: &str = "default";
+use anyhow::{ensure, Context, Result};
 
 pub(crate) fn validate_key_value_stores(key_value_stores: &Option<Vec<String>>) -> Result<()> {
-    match key_value_stores
-        .iter()
-        .flatten()
-        .find(|id| *id != ONLY_ALLOWED_KV_STORE)
-    {
-        None => Ok(()),
-        Some(invalid) => {
-            let err = anyhow!("Invalid key-value store '{invalid}'. This version of Spin supports only the '{ONLY_ALLOWED_KV_STORE}' store.");
-            Err(err)
-        }
+    for store in key_value_stores.iter().flatten() {
```

Comment:
Non-default stores can't currently be deployed to cloud, I think, so either cloud needs to reject them with a meaningful message or we need to keep the old validation on the `deploy` path.

Reply:
This really ought to be checked in cloud.

Reply:
@bacongobbler - do we have logic to check store type in Nimbus?

Reply:
This issue is just with the store _name_: https://linear.app/fermyon/issue/NIM-594/validate-app-manifest-key-value-stores-on-revision-creation

Reply:
https://github.com/fermyon/spin/pull/1389